packages/sdk/signer.ts

```diff
@@ -84,3 +86,9 @@ export type CredentialAssertion = KeyAssertion | Fido2Assertion | PasswordAssert
 export interface CredentialSigner<T extends CredentialAssertion = FirstFactorAssertion> {
   sign(challenge: UserActionChallenge): Promise<T>
 }
+
+export class EmptySigner implements CredentialSigner {
+  async sign(): Promise<never> {
+    throw new DfnsError(-1, 'A signature is required, but an EmptySigner cannot fulfill it.')
+  }
+}
```

Comment:
why do we need this?

Reply:
It's gonna a helper to be able to use the auth client for the un-authed endpoints, eg:
```ts
const dfnsClient = new DfnsApiClient({
    appId: process.env.DFNS_APP_ID!,
    baseUrl: process.env.DFNS_API_URL!,
    authToken: '',
    signer: new EmptySigner(),
  })
```

Reply:
other thought -- unrelated -- maybe we should make `baseUrl` optional (by default it's gonna be the prod url)

Reply:
in that case wouldn't it be better if `authToken` and `signer` are nullable?

re: `baseUrl` I kinda like being explicit, feels less likely that they will be hitting the wrong api (ninja vs prod).

Reply:
yeah, we can also make auth token and signer nullable, but I think original reason we did not do that, is because since 98% of cases you need those to use the api, we did not wanna wrongfully push people to "not fill those", since they wouldn't see the TS warnings. Though that's just for TS, not JS. So I'd be ok to make them nullable (not optional), and raise an error if authToken/signer is needed but not specified

re: `baseUrl` yeah ok. I think I'm thinking more towards PLG when people are just onboarded to prod. Would be nicer for them if they don't need to think about the api endpoint, just download the SDK, add an `appId` and that's it

Reply:
ok, let's make them nullable 👍

Reply:
if long term we still want to retire ninja, then I guess defaulting to prod would make sense eventually. that will be the only possible env externally at some point.

Reply:
ok updated options as optional and so removed the unecessary `EmptySigner`